aspnetcore/fundamentals/http-requests.md

```diff
@@ -0,0 +1,241 @@
+
+## Typed clients
+
+Typed Clients give you the same capabilities as named clients without the need for using strings as keys. This provides intellisense and compiler help when consuming clients. They also provide a single location to configure and interact with a particular HttpClient. For example, a single typed client might be used for a single backend endpoint and encapsulate all logic which deals with that endpoint. 
+
+A typed client is expected to accept a HttpClient via it's constructor. 
+
+```csharp
+public class GitHubService
+{
+    public HttpClient Client { get; private set; }
+
+    public GitHubService(HttpClient client)
+    {
+        client.BaseAddress = new Uri("https://api.github.com/");
+        client.DefaultRequestHeaders.Add("Accept", "application/vnd.github.v3+json"); // Github API versioning
+        client.DefaultRequestHeaders.Add("User-Agent", "HttpClientFactory-Sample"); // Github requires a user-agent
+
+        Client = client;
+    }
+}
+```
+
+To register a typed client the generic AddHttpClient method can be used, specifying our typed client class. 
+
+```csharp
+public void ConfigureServices(IServiceCollection services)
+{
+    services.AddHttpClient<GitHubService>();
+    services.AddMvc();
+}
+```
+
+The typed client is registered as transient with the DI framework.
+
+The typed client can then be injected and consumed directly. 
+
+```csharp
+public class MyController : Controller
+{
+    private GitHubService _gitHubService;
+
+    public MyController(GitHubService gitHubService)
+    {
+        _gitHubService = gitHubService;
+    }
+
+    public IActionResult Index()
+    {
+        var result = await _gitHubService.Client.GetStringAsync("/orgs/octokit/repos");
+        return Ok(result);
+    }
+}
+```
+
+In this example we only moved configuration into the type, but we could also have methods with behaviour and not actually expose the HttpClient if we want all access to go through this type. 
+
+If you prefer, the configuration for a typed client can be specified during registration, rather than in the constructor for the typed client.
+
+```csharp
+public void ConfigureServices(IServiceCollection services)
+{
+    services.AddHttpClient<GitHubService>(c =>
+    {
+        c.BaseAddress = new Uri("https://api.github.com/");
+
+        c.DefaultRequestHeaders.Add("Accept", "application/vnd.github.v3+json"); // Github API versioning
+        c.DefaultRequestHeaders.Add("User-Agent", "HttpClientFactory-Sample"); // Github requires a user-agent
+    });
+    services.AddMvc();
+}
+```
+
+If we want to entirely encapsulate the HttpClient in our typed client, rather than exposing it as a property we can define our own methods which control the client
```

Comment:
- Need a comma after "rather than exposing it as a property".
- Need a period at the end of this sentence.

Reply:
I think this can be rewritten without the first person.